### PR TITLE
New integration - CometMonitor

### PR DESCRIPTION
**Pull request by alexkuzmik**
This PR introduces a new monitoring option - `CometMonitor` which comes up as an official integration with [CometML](https://www.comet.com/site/).

The new monitor is covered with unit tests.

Notes:
* We've updated `docs/code-docs/source/monitor.rst` but it doesn't look used anymore
* We've updated the "Monitoring Module" section name in `config-json.md` to be generic so the next integration won't require updating it.

### CODE REVIEW

**Comment by alexkuzmik** [comment]
@microsoft-github-policy-service agree [company="{Comet}"]

**Comment by alexkuzmik** [comment]
Hi, @loadams! I hope I fixed the issues in my unit tests (they revealed themselves only in multi-gpu). Could you please re-run the tests?

File: requirements/requirements-dev.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 accelerate
 clang-format==16.0.2
+comet_ml>=3.41.0
```
Comment (loadams) [inline requirements/requirements-dev.txt:3]: @mrwyattii - thoughts on moving things like comet_ml, wandb, etc for monitoring to a new requirements file?

Comment (loadams) [reply requirements/requirements-dev.txt:3]: Discussed offline, this is fine.

File: docs/_pages/config-json.md [hunk 2]
```diff
@@ -1139,15 +1139,16 @@ DeepSpeed Data Efficiency Library includes two techniques: curriculum learning a
...
 {: .notice--warning}
 **Note:** Logging to WandB requires that the `wandb` package is installed (read more in the [WandB documentation](https://docs.wandb.ai/quickstart)).
 {: .notice--warning}
+**Note:** Logging to Comet requires that the `comet_ml` package is installed (read more in the [Comet documentation](https://www.comet.com/docs/v2/guides/quickstart/#1-install-and-configure-the-comet-ml-sdk)).
+{: .notice--warning}
 
-
-Deepspeed's Monitor module can log training details into a [Tensorboard](https://www.tensorflow.org/tensorboard)-compatible file, to [WandB](https://wandb.ai/site), or to simple CSV files. Below is an overview of what DeepSpeed will log automatically.
+Deepspeed's Monitor module can log training details into a [Tensorboard](https://www.tensorflow.org/tensorboard)-compatible file, to [WandB](https://wandb.ai/site), to [Comet](https://www.comet.com/site/?utm_source=deepseed&utm_medium=docs&utm_content=docs) or to simple CSV files. Below is an overview of what DeepSpeed will log automatically.
```
Comment (loadams) [inline docs/_pages/config-json.md:1151]: Worth just linking to the main comet.com site?

**Review by loadams — APPROVED** [review]

**Comment by nelyahu** [comment]
Hi @loadams @alexkuzmik ,
@deepcharm confirmed a bug in this PR when determining the config monitor.enabled flag.
see fix in https://github.com/microsoft/DeepSpeed/pull/5633

**Pull request merged**